src/test/java/org/elasticsearch/index/query/TemplateQueryParserTest.java

```diff
@@ -122,6 +122,74 @@ public void testParser() throws IOException {
         assertTrue("Parsing template query failed.", query instanceof ConstantScoreQuery);
     }
 
+    /**
+     * Test that the template query parser can parse and evaluate template expressed as a single string
+     * with conditional clause.
+     */
+    @Test
+    public void testParseTemplateAsSingleStringWithConditionalClause() throws IOException {
+        String templateString = "{" +
+                "  \"template\" : \"{ \\\"match_{{#use_it}}{{template}}{{/use_it}}\\\":{} }\"," +
+                "  \"params\":{" +
+                "    \"template\":\"all\"," +
+                "    \"use_it\": true" +
+                "  }" +
+                "}";
+
```

Comment:
@MaineC it seems I misunderstood how to formulate the requests for Template Query.

The purpose of this PR was to fix the issue with **Search Templates** in the first place but IIRC both the Template Query and Search Templates were using the same code for parsing the templates thus I added also tests for Template Query. The template parsing logic was a bit dirty IMO (my changes made it even more messy) and it seemed to me that later modifications to this logic could easily break parsing logic in Template Query without noticing it in tests. Thus I decided to add specific tests. I tried to explain why I thought these tests were useful in comments.

Looking at required request format of Template Query the string value should be provided under "query" key. So it might be useful to check why it was possible that my requests passed for me. You mentioned that you were trying this against current master. The chance is that parsing logic got changed in the mean time and does not support this exotic request format (which is probably good).

Do you want me to look at this PR again? (I can not promise I can look at this quickly.)